aten/src/ATen/native/xnnpack/Convolution.cpp

```diff
@@ -201,18 +200,38 @@ Tensor run(
       MemoryFormat::ChannelsLast,
       padded_input_nhwc.names());
 
-  const xnn_status setup_status = xnn_setup_convolution2d_nhwc_f32(
-      context.op.get(),                                      // operator
-      padded_input_nhwc.size(Layout::Activation4D::batch),   // batch_size
-      padded_input_nhwc.size(Layout::Activation4D::height),  // input_height
-      padded_input_nhwc.size(Layout::Activation4D::width),   // input_width
-      padded_input_nhwc.data_ptr<float>(),                   // input
-      output.data_ptr<float>(),                              // output
-      caffe2::pthreadpool_());                               // threadpool
-
-  TORCH_CHECK(
-      xnn_status_success == setup_status,
-      "xnn_setup_convolution2d_nhwc_f32 failed!");
+  if (context.cached_input_ptr != padded_input_nhwc.data_ptr<float>() ||
+      context.cached_output_ptr != output.data_ptr<float>() ||
+      context.batch_size !=
+        padded_input_nhwc.size(Layout::Activation4D::batch) ||
+      context.input_channels !=
+        padded_input_nhwc.size(Layout::Activation4D::channels) ||
+      context.input_height !=
+        padded_input_nhwc.size(Layout::Activation4D::height) ||
+      context.input_width !=
+        padded_input_nhwc.size(Layout::Activation4D::width)
+      ) {
```

Comment:
_Personal_ opinion:

1) Consider using parentheses liberally.
2) I try to put rvalues (or more pedantically prvalues) on the left side of comparisons.  That way `==` or `!=` misspelled as `=` triggers a compiler error.  Otherwise it is a warning, but it is typical in codebases where a 'warning as error' policy is not enforced to fall into the habit of ignoring warnings.

Please feel free to ignore.  Both of these the pedantic side.

Reply:
I strongly agree withe second one. I will make those changes. I weakly agree with first one :).